internal/metrics/metrics.go

```diff
@@ -0,0 +1,174 @@
+package metrics
+
+import (
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	lokiv1beta1 "github.com/ViaQ/loki-operator/api/v1beta1"
+)
+
+const (
+	CompactorEnabled  string = "compactor_enabled"
+	CompactorDisabled string = "compactor_disabled"
+	TenantLimits      string = "tenant_limits"
+	GlobalLimits      string = "global_limits"
+)
+
+var (
+	sizeTracker = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "lokistack_cluster_sizes",
+			Help: "Number of clusters that are using a specific size configuration",
+		},
+		[]string{"size"},
+	)
+
+	globalStreamTracker = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "lokistack_max_global_streams",
+			Help: "Total of max global streams defined by the custom resource",
+		},
+		[]string{"size"},
+	)
+
+	tenantStreamTracker = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "lokistack_avg_tenant_streams",
+			Help: "Average number of streams for all tenants defined by the custom resource",
+		},
+		[]string{"size", "tenants"},
+	)
+
+	compactorUsageTracker = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "lokistack_compactor_enablement",
+			Help: "Number of clusters that use the compactor component",
+		},
+		[]string{"size", "state"},
+	)
+
+	limitsUsageTracker = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "lokistack_limits_usage",
+			Help: "Number of clusters that are using global or tenant limits",
+		},
+		[]string{"size", "type"},
+	)
+
+	metricCollectors = []prometheus.Collector{
+		sizeTracker,
+		limitsUsageTracker,
+		compactorUsageTracker,
+		globalStreamTracker,
+		tenantStreamTracker,
+	}
+)
+
+// Registers the prometheus collectors with the k8 default metrics
+func RegisterMetricCollectors() {
+	for _, collector := range metricCollectors {
+		metrics.Registry.MustRegister(collector)
+	}
+}
+
+// Sets the metrics for the cluster size
+func SetSizeMetric(size lokiv1beta1.LokiStackSizeType) {
+	sizeTypes := []lokiv1beta1.LokiStackSizeType{
+		lokiv1beta1.OneXExtraSmallSize,
+		lokiv1beta1.OneXSmall,
+		lokiv1beta1.OneXMedium,
+	}
+
+	for _, sizeType := range sizeTypes {
+		var isSize float64 = 0
+
+		if sizeType == size {
+			isSize = 1
+		}
+
+		sizeTracker.With(prometheus.Labels{
+			"size": SizeString(size),
+		}).Set(isSize)
+	}
```

Comment:
The check for a size type is not idiomatic for golang. Consider the following:
```go
var value string
switch sizeType {
case lokiv1beta1.OneXExtraSmallSize:
  value = "1x.xsmall"
case lokiv1beta1.OneXSmall:
  value = "1x.small"
case lokiv1beta1.OneXMedium:
  value = "1x.medium"
}

sizeMetric.With(prometheus.Labels{"size": value}).Set(1)
```

In addition we do not need to register the zero values for a type that is not given. Each Spec can have only one Size type and thus we want to register this only.